packages/elements-core/src/utils/exampleGeneration.ts

```diff
@@ -6,21 +6,25 @@ import React from 'react';
 
 import { useDocument } from '../context/InlineRefResolver';
 
+export type GenerateExampleFromMediaTypeContentOptions = Sampler.Options;
+
 export const useGenerateExampleFromMediaTypeContent = (
   mediaTypeContent: IMediaTypeContent | undefined,
   chosenExampleIndex?: number,
+  options?: GenerateExampleFromMediaTypeContentOptions,
 ) => {
   const document = useDocument();
   return React.useMemo(
-    () => generateExampleFromMediaTypeContent(mediaTypeContent, document, chosenExampleIndex),
-    [mediaTypeContent, document, chosenExampleIndex],
+    () => generateExampleFromMediaTypeContent(mediaTypeContent, document, chosenExampleIndex, options),
+    [mediaTypeContent, document, chosenExampleIndex, options],
```

Comment:
I see...
Well unfortunately the objects pattern doesn't play nice with hooks, but you still can't expect consumers to just memoize them all the time I think. In this case the workaround is probably to split up the options, add them all to the dependency array, then reassemble it when calling the function.

Reply:
That... that sounds horrible 😂

Reply:
That is